packages/analyzer/cli.js

```diff
@@ -0,0 +1,135 @@
+import globby from 'globby';
+import fs from 'fs';
+import commandLineArgs from 'command-line-args';
+import chokidar from 'chokidar';
+import debounce from 'debounce';
+
+import { create } from './src/create.js';
+import {
+  getUserConfig,
+  getCliConfig,
+  addFrameworkPlugins,
+  addCustomElementsPropertyToPackageJson,
+  mergeGlobsAndExcludes,
+  timestamp,
+  DEFAULTS,
+  MENU,
+} from './src/utils/cli-helpers.js';
+import { findExternalManifests } from './src/utils/find-external-manifests.js';
+
+/**
+ * @param {{argv:string[]; cwd: string}} opts
+ */
+export async function cli({ argv = process.argv, cwd = process.cwd() }) {
+  const mainDefinitions = [{ name: 'command', defaultOption: true }];
+  const mainOptions = commandLineArgs(mainDefinitions, { stopAtFirstUnknown: true, argv });
+  const cliArgs = mainOptions._unknown || [];
+
+  if (mainOptions.command === 'analyze') {
+    const { config: configPath, ...cliConfig } = getCliConfig(cliArgs);
+    const userConfig = await getUserConfig(configPath, cwd);
+
+    /**
+     * Merged config options
+     * Command line options override userConfig options
+     */
+    const mergedOptions = { ...DEFAULTS, ...userConfig, ...cliConfig };
+    const merged = mergeGlobsAndExcludes(DEFAULTS, userConfig, cliConfig);
+    const globs = await globby(merged, { cwd });
+
+    async function run() {
+      const modules = userConfig?.overrideModuleCreation
+        ? userConfig.overrideModuleCreation({ ts, globs })
+        : globs.map((glob) => {
+            const fullPath = path.resolve(cwd, glob);
+            const source = fs.readFileSync(fullPath).toString();
+
+            return ts.createSourceFile(glob, source, ts.ScriptTarget.ES2015, true);
+          });
+
+      /**
+       * @TODO
+       * filter out thirdPartyCEMs that are not dependencies
+       * [{isDependency: true, name: 1}, {isDependency: true, name: 2}, {name: 3}].filter(({isDependency}) => !isDependency)
+       */
+      let thirdPartyCEMs = [];
+      if (mergedOptions?.dependencies) {
+        try {
+          const fullPathGlobs = globs.map(glob => path.resolve(cwd, glob));
+          thirdPartyCEMs = await findExternalManifests(fullPathGlobs, {basePath: cwd});
+          /** Flatten found CEMs, and mark them as being a dependency so we can remove them later */
+          thirdPartyCEMs
+            .flatMap(({ modules }) => modules)
+            .map((mod) => ({ ...mod, isDependency: true }));
+
+          if (mergedOptions.dev) console.log(thirdPartyCEMs);
+        } catch (e) {
+          if (mergedOptions.dev) console.log(`Failed to add third party CEMs. \n\n${e.stack}`);
+        }
+      }
+
+      let plugins = await addFrameworkPlugins(mergedOptions);
+      plugins = [...plugins, ...(userConfig?.plugins || [])];
+
+      /**
+       * Create the manifest
+       */
+      const customElementsManifest = create({
+        modules,
+        plugins,
+        dev: mergedOptions.dev,
+        thirdPartyCEMs,
```

Comment:
I think instead of passing `dev` and `thirdPartyCEMs` to `create`, I think we should initialize the `context` object above, so something like:

```js
const context = { dev, thirdPartyCEMs };
const customElementsManifest = create({modules, plugins, context});
```

Reply:
and then remove creation of the `context` object in `create.js`